railties/lib/rails/generators/app_base.rb

```diff
@@ -171,6 +175,12 @@ def database_gemfile_entry
                             "Use #{options[:database]} as the database for Active Record"
       end
 
+      def webserver_gemfile_entry
+        return [] if options[:skip_puma]
+        comment = 'Use Puma as the app server'
+        GemfileEntry.new('puma', nil, comment)
```

Comment:
We will when my changes to puma https://github.com/puma/puma/pull/856 get merged. For now i don't think we're relying on any specific version feature. The previous `Gemfile` declaration did not list a version.